fenix/automation/taskcluster/androidTest/testrail.py

```diff
@@ -0,0 +1,103 @@
+
+def build_milestone_description(milestone_name):
+    current_date = datetime.now()
+    formatted_date = current_date = current_date.strftime("%B %d, %Y")
+    return textwrap.dedent(f"""
+        RELEASE: {milestone_name}\n\n\
+        RELEASE_TAG_URL: https://github.com/mozilla-mobile/fenix/releases\n\n\
+        RELEASE_DATE: {formatted_date}\n\n\
+        TESTING_STATUS: [ TBD ]\n\n\
+        QA_RECOMMENDATION:[ TBD ]\n\n\
+        QA_RECOMENTATION_VERBOSE: \n\n\
+        TESTING_SUMMARY\n\n\
+        Known issues: n/a\n\
+        New issue: n/a\n\
+        Verified issue: 
+    """)
+
+class TestRail():
+
+    def __init__(self):
+        try:
+            self.client = APIClient(TESTRAIL_HOST)
+            self.client.user = TESTRAIL_USERNAME
+            self.client.password = TESTRAIL_PASSWORD
+        except KeyError as e:
+            raise ValueError(f"ERROR: Missing Testrail Env Var: {e}")
+    
+    # Public Methods
+
+    def create_milestone(self, testrail_project_id, title, description):
+        data = {"name": title, "description": description}
+        return self.client.send_post(f'add_milestone/{testrail_project_id}', data)
+    
+    def create_test_run(self, testrail_project_id, testrail_milestone_id, name_run, testrail_suite_id):
+        data = {"name": name_run, "milestone_id": testrail_milestone_id, "suite_id": testrail_suite_id}
+        return self.client.send_post(f'add_run/{testrail_project_id}', data)
+    
+    def update_test_cases_to_passed(self, testrail_project_id, testrail_run_id, testrail_suite_id):
+        test_cases = self._get_test_cases(testrail_project_id, testrail_suite_id)
+        data = { "results": [{"case_id": test_case['id'], "status_id": 1} for test_case in test_cases]}
+        return testrail._update_test_run_results(testrail_run_id, data)
+
+    # Private Methods
+
+    def _get_test_cases(self, testrail_project_id, testrail_test_suite_id):
+        return self.client.send_get(f'get_cases/{testrail_project_id}&suite_id={testrail_test_suite_id}')
+    
+    def _update_test_run_results(self, testrail_run_id, data):
+        return self.client.send_post(f'add_results_for_cases/{testrail_run_id}', data)
+    
+if __name__ == "__main__":
+    if TEST_STATUS != 'PASS':
+        raise ValueError("Tests failed. Sending Slack Notification....")
+
+    PROJECT_ID = 53 # Firefox for FireTV
+    TEST_SUITE_ID = 45442 # Demo Test Suite
+
+    testrail = TestRail()
+    milestone_name = build_milestone_name(PRODUCT_TYPE, RELEASE_TYPE, parse_release_number(VERSION_NUMBER))
+    milestone_description = build_milestone_description(milestone_name)
+
+    # Create milestone for 'Firefox for FireTV' and store the ID
+    milestone_id = testrail.create_milestone(PROJECT_ID, milestone_name, milestone_description)['id']
+
+    # Create test run for each Device/API and update test cases to 'passed'
+    for test_run_name in ['Google Pixel 32(Android11)', 'Google Pixel2(Android9)']:
```

Comment:
discuss with oana if these are likely to change, should these be params?

Reply:
Yes, they will be changing.